programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -132,6 +132,101 @@ pub enum SyscallError {
 
 type Error = Box<dyn std::error::Error>;
 
+pub trait HasherImpl {
+    const HASH_BYTES: usize;
+    const NAME: &'static str;
+
+    fn create_hasher() -> Self;
+    fn hash(&mut self, val: &[u8]);
+    fn result(self) -> [u8; HASH_BYTES];
```

Comment:
```suggestion
    fn result(self) -> [u8; Self::HASH_BYTES];
```
(and updates throughout)

otherwise i believe the outter scoped `hash::HASH_BYTES` is in scope

Reply:
so I tried updating it but then started getting this error "generic parameters may not be used in const operations" and "type parameters may not be used in const expressions" So the only 2 solutions I could think of is either go with constant numerical value '32' or just use outer scoped hash::HASH_BYTES for trait only and for the implementations, can use 'Self::HASH_BYTES'.

Reply:
oh that's probably from the array definition :thinking: 

maybe instead of the associated constant on the trait. declare an associated _type_?

something like
```
type Output: AsRef<[u8]>;
```

`AsRef<u8>` should be enough for the `copy_from_slice()` we do to send the result back to vm.

this method then becomes
```
fn result(self) -> Self::Output;
```

Reply:
Done!